doc-deploy-stable/action.yml

```diff
@@ -1,164 +1,286 @@
 name: "Documentation deploy stable version."
-description: "Deploy stable documentation to GitHub pages."
+description: "Deploy stable documentation artifacts to desired repository branch."
 
 inputs:
   doc-artifact-name:
     description: "Name of the HTML documentation artifact."
     required: false
     default: 'documentation-html'
     type: string
-  decompress_artifact:
+  decompress-artifact:
     description: "If ``True``, the documentation artifact name is decompressed. If ``False``, content is assumed to be a collection of HTML files."
     required: false
     default: false
     type: string
+  repository:
+    description: "Name of the repository in the form of ``username/repository`` used to deploy the documentation. If not specified, current repository is used."
+    required: false
+    default: 'current'
+    type: string
+  branch:
+    description: "Desired deployment branch name. If not specified, ``gh-pages`` is used."
+    required: false
+    default: 'gh-pages'
+    type: string
   cname:
     description: 'The canonical name of the documentation website.'
     required: true
     type: string
-  version-config:
-    description: 'Name of the JSON file specifying version configuration.'
-    default: 'versions.json'
-    required: false
-    type: string
   token:
     description: "Required token for documentation deployment."
     required: true
     type: string
-  python-version:
-    description: 'Desired Python version.'
-    default: '3.10'
+  commit-message:
+    description: "Commit message used when deploying the documentation"
     required: false
+    default: 'DOC: update development documentation'
     type: string
   render-last:
     description: 'Number of last versions to be rendered in multi-doc.'
     default: '3'
     required: false
     type: string
-  checkout:
-      description: 'Whether to checkout the repository or not.'
-      default: true
-      required: false
-      type: boolean
 
 runs:
   using: "composite"
   steps:
 
-    - name: "Checkout project in the GitHub Pages branch"
-      uses: actions/checkout@v3
-      if: ${{ inputs.checkout == 'true' }}
-      with:
-        ref: 'gh-pages'
+    # Verify that a tag event triggered the workflow. This is a strong requirement to
+    # guarantee that any stable documentation deployment is linked to a tag.
+    # If a tag is detected, verify that it is compliant with semantic version syntax.
```

Comment:
I am struggling with this requirement...

This would force us to do a new release if we just wanted to update the ``ansys_sphinx_theme`` version on a release branch. For me this shouldn't be necessary. You could just update the workflow, update the theme version and redeploy the docs. The theme is not a strict requirement of the package and it shouldn't need the deployment of a new version...

I'd suggest that this "tag exists" check wasn't performed. I'd check though that you at least where on a ``release/*`` branch. We've already seen repos with bad practices in which they directly tag on main (which is not a good practice).

Reply:
Thanks for starting this discussion, @RobPasMue.

When I implemented this requirement, I knew it was very strict and was not complete satisfied by it.

While testing this, I accidentally created a tag `v0.4` in a `release/0.3` branch. With the current logic of the actions, those generated a new version based on the tag and not on the branch name.

Hence, I agree about giving preference to the branch name rather than the tag.

Reply:
This is a very good point.

Reply:
The GitHub context always gives preference to tags over branches. This translates into different variables in the actions context.

github.ref_name | string | The short ref name of the branch or tag that triggered the workflow run. This value matches the branch or tag name shown on GitHub. For example, feature-branch-1.
-- | -- | --

Hence, I implemented to ways of extracting the version:

* From a `tag` the action drops the "v" prefix and verifies that the version number follows semantic versioning.
* From a `branch`, the action drops the "release/" prefix and verifies that the version number follows "X.Y" semantic versioning.

Reply:
Sounds good to me! Thanks @jorgepiloto!